src/python_testing/TC_AVSUM_2_4.py

```diff
@@ -0,0 +1,211 @@
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --endpoint 1
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main, has_cluster, run_if_endpoint_matches
+from mobly import asserts
+from TC_AVSUMTestBase import AVSUMTestBase
+
+
+class TC_AVSUM_2_4(MatterBaseTest, AVSUMTestBase):
+    has_feature_dptz = False
+    has_feature_mpan = False
+    has_feature_mtilt = False
+    has_feature_mzoom = False
+    has_feature_mpresets = False
+
+    def ptz_range_validation(self, mptz, tiltmin, tiltmax, panmin, panmax, zoommax):
+        if self.has_feature_mpan:
+            if mptz.pan is None:
+                asserts.fail("Pan value missing when MPAN feature set")
+            else:
+                apan = mptz.pan
+                asserts.assert_greater_equal(apan, panmin, "Pan out of range of DUT defined values")
+                asserts.assert_less_equal(apan, panmax, "Pan out of range of DUT defined values")
+
+        if self.has_feature_mtilt:
+            if mptz.tilt is None:
+                asserts.fail("Tilt value missing when MTILT feature set")
+            else:
+                atilt = mptz.tilt
+                asserts.assert_greater_equal(atilt, tiltmin, "Tilt out of range of DUT defined values")
+                asserts.assert_less_equal(atilt, tiltmax, "Tilt out of range of DUT defined values")
+
+        if self.has_feature_mzoom:
+            if mptz.zoom is None:
+                asserts.fail("Zoom value missing when MZOOM feature set")
+            else:
+                azoom = mptz.zoom
+                asserts.assert_greater_equal(azoom, 1, "Zoom out of range of DUT defined values")
+                asserts.assert_less_equal(azoom, zoommax, "Zoom out of range of DUT defined values")
+
+    def verify_preset_matches(self, preset, position):
+        if self.has_feature_mpan:
+            asserts.assert_equal(preset.settings.pan, position.pan)
+
+        if self.has_feature_mtilt:
+            asserts.assert_equal(preset.settings.tilt, position.tilt)
+
+        if self.has_feature_mzoom:
+            asserts.assert_equal(preset.settings.zoom, position.zoom)
+
+    def desc_TC_AVSUM_2_4(self) -> str:
+        return "[TC-AVSUM-2.4] Attributes with DUT as Server"
+
+    def steps_TC_AVSUM_2_4(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Verify that the MPRESETS feature is supported, exit otherwise"),
```

Comment:
Rather than doing this, suggest using the feature PICS code and has_feature on the decorator. Right now, this test will run for all devices that have this cluster, and at the test event it will look like you have a bunch of test runs, but they won't do anything and then the ATLs will (hopefully) find that and say it's weird.